src/main/java/br/com/deveficiente/bolaoapi/services/championship/Match.java

```diff
@@ -86,8 +83,37 @@ public boolean hasConflict(Match other) {
         return matchTeams;
     }
 
+    public void addShot(@Valid Shot shot) {
+        Assert.isTrue(!shots.contains(shot), "Shot already made for this match.");
+        Assert.isTrue(!shot.getDoubled() || !hasDoubledShotInRound(shot), "Only one doubled shot is permitted per round.");
```

Comment:
Essa primeira negação, qual o motivo mesmo? pq você nega o retorno do shot ser dobrado?

Reply:
Pq se não for um caso de palpite dobrado, eu não preciso verificar se já existe um dobrado na rodada. 

Apenas para evitar a verificação no caso de palpites normais.